examples/expressions/13_timeseries_forecasting.py

```diff
@@ -0,0 +1,527 @@
+
+"""
+
+# %%
+# A problem with relevant datetime features
+# -----------------------------------------
+#
+# We use a dataset of bike sharing demand in 2011 and 2012.
+# In this setting, we want to predict the number of bike rentals, based
+# on the date, time and weather conditions.
+# In this example, we use Polars dataframes instead of Pandas.
+
+
+import polars as pl
+
+from skrub import datasets
+
+data = datasets.fetch_bike_sharing().bike_sharing
+data = pl.from_pandas(data)
+
+# %%
+# As first step, we use the skrub |TableReport| to explore the data and get an idea
+# of the features we are working with.
+# In particular, we can see that the dataset contains a "date" column, and that
+# the "cnt" column is the target we want to predict. We can get a high level
+# overview of the features in the dataset by checking out the ``Stats`` tab in the
+# report. Among other things, we can see that the "date" column is a string column.
+# This is potentially problematic, and a good use case for the skrub |DatetimeEncoder|,
+# as we show in the rest of the example.
+
+from skrub import TableReport
+
+TableReport(data)
+
+
+# %%
+# Prediction with datetime features
+# ---------------------------------
+#
+# In this section, we use the skrub |DatetimeEncoder|, paired with the skrub
+# Expressions to build a predictive pipeline
+# that pre-processes the data, performs feature engineering and executes
+# hyperparameter optimization to improve the prediction performance.
+# We will use a |RidgeCV| model as our learner. While it is not the most powerful
+# model, it will help us illustrate how feature engineering affects the prediction
+# performance.
+
+
+from sklearn.linear_model import RidgeCV
+
+# %%
+# Since we are working with timeseries, we need to use |TimeSeriesSplit| to perform
+# crossvalidation.
+from sklearn.model_selection import TimeSeriesSplit
+from sklearn.pipeline import make_pipeline
+
+ts = TimeSeriesSplit(
+    n_splits=3,  # to keep the notebook fast enough on common laptops
+    gap=48,  # 2 days (48 hours) data gap between train and test
+    max_train_size=10000,  # keep train sets of comparable sizes
+    test_size=3000,  # for 2 or 3 digits of precision in scores
+)
+
+
+# %%
+# We import ``skrub`` to access the skrub expressions and define a skrub |var|
+# to start with. We use |.skb.subsample| to reduce the size of the dataset used
+# in the preview to 100 rows, so that the notebook runs faster when developing.
+# When the models are trained, the expressions switch automatically to the full
+# dataset.
+
+import skrub
+
+data_var = skrub.var("data", data).skb.subsample(n=100)
+
+# We extract our input data (``X``) and the target column (``y``), and mark them
+# as X and y.
+X = data_var[
+    "date", "holiday", "weathersit", "temp", "hum", "windspeed"
+].skb.mark_as_X()
+y = data_var["cnt"].skb.mark_as_y()
+X
+
+
+# %%
+# We can use the |Cleaner| to clean the data and convert the date column to a
```

Comment:
"Clean the data" is too vague; could you list what this transformer is doing in this instance?